GitTfs/Commands/Fetch.cs

```diff
@@ -120,15 +120,14 @@ private void FetchRemote(bool stopOnFailMergeCommit, IGitTfsRemote remote)
 
         protected virtual void DoFetch(IGitTfsRemote remote, bool stopOnFailMergeCommit)
         {
-            if (remote.Repository.IsBare)
+            var bareBranch = string.IsNullOrEmpty(BareBranch) ? remote.Id : BareBranch;
+            if (!ForceFetch &&
+                remote.Repository.IsBare &&
+                remote.Repository.HasRef(GitRepository.ShortToLocalName(bareBranch)) &&
+                remote.MaxCommitHash != remote.Repository.GetCommit(bareBranch).Sha)
             {
-                if(string.IsNullOrEmpty(BareBranch))
-                    throw new GitTfsException("error : specify a git branch to fetch on...");
-                if (!remote.Repository.HasRef(GitRepository.ShortToLocalName(BareBranch)))
-                    throw new GitTfsException("error : the specified git branch doesn't exist...");
-                if (!ForceFetch && remote.MaxCommitHash != remote.Repository.GetCommit(BareBranch).Sha)
-                    throw new GitTfsException("error : fetch is not allowed when there is ahead commits!",
-                        new List<string>() {"Remove ahead commits and retry", "use the --force option (ahead commits will be lost!)"});
+                throw new GitTfsException("error : fetch is not allowed when there is ahead commits!",
```

Comment:
Something makes me feel that we do not need this check at all. OR, at least, we need to try to stabilize tfs remote and git branch first.